src/cli/java/org/commcare/util/screen/EntityListSubscreen.java

```diff
@@ -236,15 +236,17 @@ public boolean handleInputAndUpdateHost(String input, EntityScreen host, boolean
             try {
                 int index = Integer.parseInt(input);
                 host.setHighlightedEntity(mChoices[index]);
+                host.setCurrentScreenToDetail();
```

Comment:
Would this not result into moving to the Details screen directly without showing the Entity List screen in the normal workflow ?

Reply:
In the CLI? At this point we've already shown the entity list screen, so this tells the CLI to show details when it redraws the screen.

Reply:
Sorry I meant the question for Web Apps UI, mostly trying to understand if this change can impact the web apps case list -> detail navigation.

Reply:
I don't think it matters for web apps, because web apps uses the `get_details` endpoint, so I don't think it ever shows details except in response to `get_details`...but I don't know a lot about the `get_details` workflow, let me verify that.

Reply:
Looked at the code a bit and am more confident: the only place formplayer creates an `EntityDetailListResponse` is when responding to the specific `get_details` request, in `MenuController.getDetails`. In response to navigation requests, `advancedSessionWithSelections` returns either a form entry response or the return value of `getNextMenu`, which when it gets an EntityScreen doesn't look at the current subscreen, it always sends back an EntityListResponse: https://github.com/dimagi/formplayer/blob/2c916265655569583a5a397dc2c98912196a8379/src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java#L132-L153

Reply:
Makes sense to me, thanks for double checking this!